Update nanoleaf documentation with examples (#16537)
home-assistant/home-assistant.github.io: source/_integrations/nanoleaf.markdown

### Configuration Sample

> The `nanoleaf` platform allows you to control [Nanoleaf Light Panels](https://nanoleaf.me) from Home Assistant. Note that for full control of nanoleaf devices, particularly effects, make sure they are set up as `- platform: nanoleaf` not as a HomeKit device.

Reviewer: ```suggestion
The Nanoleaf integration allows you to control [Nanoleaf Light Panels](https://nanoleaf.me) from Home Assistant. Note that for full control of Nanoleaf devices, particularly effects, make sure they are set up as instructed on this page and not as a HomeKit device.
```

Additionally, I think the first sentence should be moved above the first heading.

  - platform: nanoleaf
    host: 192.168.1.10
    token: xxxxxxxxxxxxxxxxxxxxx

>     name: bedroom_triangles

Reviewer 2: ```suggestion
```
We always try to keep the first example as minimal as possible, so only the required options are used.

Author: Ok, makes sense. I've removed the name from the first example, and made a second example with the name, but it's pretty obvious so I can drop it entirely if it's not adding any value.

4. The output should include the auth token like *{"auth_token":"xxxxxxxxxxxxxxxxxxxxx"}*, copy the resulting token into your configuration

If you get a 403 Forbidden message, you probably did not press the *ON* button long enough. The time-frame to get a valid token is only 30 seconds, so you have to be quick to issue the curl request.

### Getting useful values for automations

First set up some effects with the Nanoleaf app, and have them installed on the device.

> Once installed, go to "Developer Tools" and find the state of the device, eg `light.bedroom_triangles`. You should end up with something like this:

Reviewer: Why does one need the developer tools? The effects are available on the light entity itself, right?

```suggestion
Once installed, go to "Developer Tools" and find the state of the device, e.g., `light.bedroom_triangles`. You should end up with something like this:
```

```yaml

> # Example nanoleaf state:

Reviewer: Dropping states in documentation causes confusion in general, as people try to use them. Please remove this blob.

Author: Ok, can do. However when trying to set up my Nanoleaf I had quite a bit of confusing working out how to find the names of the effects that I could use and how to call them, I had to read the integration code just to work out what it was expecting. This is why I have made the examples of how to find the effects from the state and then call the service to set the lights later.

Is there a better way to communicate this? At the moment the page feels like "here's how to create the light entity" but with no info on how to actually use the entity that was created.

Reviewer: If you click on a light, it just shows them...

![image](https://user-images.githubusercontent.com/195327/107847623-6f7be780-6ded-11eb-822b-67f3d8a537a0.png)

Author: Ok, that's good. Can I put a screenshot in this document? I'm not sure how I can do that.

Reviewer: Yes, however, the below concerns are in the same region at this one.

It is general Home Assistant functionality, and not specific to this integration.

min_mireds: 154
max_mireds: 833
effect_list:
  - Beatdrop
  - Bedtime
  - Blaze
  - Cocoa Beach
  - Cotton Candy
  - Date Night
  - Hip Hop
  - Hot Sauce
  - Jungle
  - Lightscape
  - Morning Sky
  - Northern Lights
  - Pop Rocks
  - Prism
  - Starlight
  - Sundown
  - Waterfall
  - Vibrant Sunrise
brightness: 155
effect: Bedtime
friendly_name: Triangles
icon: 'mdi:triangle-outline'
supported_features: 55
```

Now you can call the `light.turn_on` service with the data:
```yaml
# Example data to activate a nanoleaf effect:
entity_id: light.trianglestest
effect: Starlight

> ```

Reviewer: I feel this is out of place in this document. This is the integration documentation, not the documentation to teach how to use lights in Home Assistant.

While I get your goal, in theory, this example can be copied into hundreds of places, which is not maintainable.

Author: Yeah, removed this snippet.

Or in a script for example:
```yaml
# Example alarm script
alias: Alarm Sequence
sequence:
  - service: light.turn_on
    data:
      effect: Vibrant Sunrise
    entity_id: light.triangles
mode: single

> ```

Reviewer: same comment as above.

Author: I feel this is worth keeping, although I've added some extra commenting explaining why it exists, showing how we can use the effects that we discover from the state/UI.

Reviewer: I don't agree. It is general light control, and not anything that is specific this integration. This does not belong on this page, as controlling a light is a generic thing, not a Nanoleaf thing.

Author: Ok, I have added a disclaimer noting the versions that were used for the examples, along with a link to the proper "Lights" integration page. This should direct users to the full docs for lights to find all the info that is available, but still give them something to get started with.

4. The output should include the auth token like *{"auth_token":"xxxxxxxxxxxxxxxxxxxxx"}*, copy the resulting token into your configuration

If you get a 403 Forbidden message, you probably did not press the *ON* button long enough. The time-frame to get a valid token is only 30 seconds, so you have to be quick to issue the curl request.

### Using the Nanoleaf Device

The nanoleaf device is a standard light, so can be used with standard Home Assistant Light services for automations and scripts. For full details see the [Light Integrations Page.](/integrations/light/)

### Getting Useful Values for Automations and Scripts

> These examples were written for version 2021.2 with a Nanoleaf Shapes device on version 5.0.0 of firmware, so newer versions may have introduced or changed functionality.

Reviewer: We don't keep version history in our documentation. It always reflects the latest stable version of Home Assistant.

First set up some effects with the Nanoleaf app, and have them installed on the device.

Once installed, go to "Developer Tools" and find the state of the device, e.g., `light.bedroom_triangles`. This will help you discover the names of the effects that you can use with the device. You should end up with something like this:

```yaml
# Example nanoleaf state:
# Note that this is all device specific, you will need to look up your own device's state
min_mireds: 154
max_mireds: 833
effect_list:
  # These effects are device specific, and have been installed through the nanoleaf app
  - Beatdrop
  - Sundown
  - Waterfall
  - Vibrant Sunrise
brightness: 155
effect: Bedtime
friendly_name: Triangles
icon: 'mdi:triangle-outline'
supported_features: 55
```

Now you can call the `light.turn_on` service in a script for example:
```yaml
# Example alarm script
alias: Alarm Sequence
sequence:
  - service: light.turn_on
    data:
      # Note - this is one of the effects we discovered in the state
      effect: Vibrant Sunrise
    entity_id: light.bedroom_triangles
mode: single

> ```

Reviewer: This example should not be in this part of the documentation, as it is not specific or unique for Nanoleaf.

Author: Ok, dropped all the examples.
